datahub-web-react/src/app/applications/ApplicationsTable.tsx

```diff
@@ -0,0 +1,220 @@
+                            applicationUrn={application.urn}
+                            displayName={displayName}
+                            searchQuery={searchQuery}
+                        />
+                    );
+                },
+                sorter: (sourceA, sourceB) => {
+                    const nameA = entityRegistry.getDisplayName(EntityType.Application, sourceA.entity);
+                    const nameB = entityRegistry.getDisplayName(EntityType.Application, sourceB.entity);
+                    return nameA.localeCompare(nameB);
+                },
+                sortOrder: sortedInfo.columnKey === 'tag' ? sortedInfo.order : null,
+            },
+            {
+                title: 'Description',
+                key: 'description',
+                render: (record) => {
+                    return (
+                        <ApplicationDescriptionColumn
+                            key={`description-${record.entity.urn}`}
+                            applicationUrn={record.entity.urn}
+                            description={record.entity.properties?.description}
+                        />
+                    );
+                },
+            },
+            {
+                title: 'Owners',
+                key: 'owners',
+                render: (record) => {
+                    return (
+                        <ApplicationOwnersColumn
+                            key={`owners-${record.entity.urn}`}
+                            applicationUrn={record.entity.urn}
+                            owners={record.entity.ownership}
+                        />
+                    );
+                },
+            },
+            {
+                title: '',
+                key: 'actions',
+                alignment: 'right' as AlignmentOptions,
+                render: (record) => {
+                    return (
+                        <ApplicationActionsColumn
+                            applicationUrn={record.entity.urn}
+                            onDelete={() => {
+                                showDeleteConfirmation(record.entity.urn);
+                            }}
+                        />
+                    );
+                },
+            },
+        ],
+        [entityRegistry, searchQuery, sortedInfo, showDeleteConfirmation],
+    );
+
+    // Generate table data once with memoization
+    const tableData = useMemo(() => {
+        return filteredApplications.map((application) => ({
+            ...application,
+            key: application.entity.urn,
+        }));
+    }, [filteredApplications]);
+
+    return (
+        <>
+            <Table
+                columns={columns}
+                data={tableData}
+                isLoading={isLoading}
+                isScrollable
+                rowKey="key"
+                onChange={handleTableChange as any}
+            />
+
+            {/* Delete confirmation modal - simplified */}
+            <Modal
```

Comment:
nit: we do have a `ConfirmationModal` shared component you can use here

Reply:
addressed!